core/src/test/scala/quasar/destination/s3/S3DestinationSpec.scala

```diff
@@ -97,10 +97,10 @@ object S3DestinationSpec extends EffectfulQSpec[IO] {
 }
 
 final class MockUpload(status: Ref[IO, Map[ObjectKey, String]]) extends Upload[IO] {
-  def upload(bytes: Stream[IO, Byte], bucket: Bucket, key: ObjectKey): IO[Unit] =
+  def upload(bytes: Stream[IO, Byte], bucket: Bucket, key: ObjectKey): Stream[IO, Unit] =
     for {
-      data <- bytes.through(text.utf8Decode).compile.string
-      _ <- status.update(_ + (key -> data))
+      data <- bytes.through(text.utf8Decode)
+      _ <- Stream.eval(status.update(_ + (key -> data)))
```

Comment:
…which in turn will mean that your `Map[ObjectKey, String]` will only contain *part* of the written data (specifically, the last `String` written).

Reply:
How can I reference to the entire results from `bytes.through(text.utf8Decode)` without `flatMap`ping? Disregarding `compile` of course.